pkg/asset/agent/joiner/addnodesconfig.go

```diff
@@ -71,10 +87,44 @@ func (a *AddNodesConfig) Load(f asset.FileFetcher) (bool, error) {
 	}
 
 	params := &Params{}
-	if err := json.Unmarshal(file.Data, params); err != nil {
+	if err = json.Unmarshal(file.Data, params); err != nil {
 		return false, fmt.Errorf("failed to unmarshal %s: %w", addNodesParamsFile, err)
 	}
-
 	a.Params = *params
+
+	// Load nodes-config.yaml file.
+	file, err = f.FetchByName(nodesConfigFilename)
+	if err != nil {
+		return false, fmt.Errorf("failed to load %s file: %w", nodesConfigFilename, err)
+	}
+
+	config := Config{}
+	if err = yaml.Unmarshal(file.Data, &config); err != nil {
+		return false, fmt.Errorf("failed to unmarshal %s: %w", nodesConfigFilename, err)
+	}
+	a.Config = config
+
+	err = a.finish()
+	if err != nil {
+		return false, err
+	}
 	return true, nil
 }
+
+func (a *AddNodesConfig) finish() error {
+	if err := a.validateHosts().ToAggregate(); err != nil {
+		return errors.Wrapf(err, "invalid nodes configuration")
+	}
+	return nil
+}
+
+func (a *AddNodesConfig) validateHosts() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if len(a.Config.Hosts) == 0 {
+		fieldPath := field.NewPath("hosts")
+		allErrs = append(allErrs, field.Required(fieldPath, "at least one host must be defined"))
+	}
```

Comment:
Will complete validation of the hosts info in the file be done here, including `staticNetworkConfigGenerator.ValidateStaticConfigParams` or will that be done somewhere else?

Reply:
Thanks @bfournie for raising up this point, on a second view the NMStateConfig asset if effectively required, I fixed it here https://github.com/openshift/installer/pull/8093/commits/b4c8aed7aa1930aa8c1fe28b5bbda7b3c5249195